File: Engineering concepts/Design Patterns and Principles/SingletonPatternExample/Logger.java
public class Logger {

    // Single instance of Logger
    private static Logger instance = new Logger();

    // Private constructor
    private Logger() {
        System.out.println("Logger initialized successfully.");
    }

    // Global access point
    public static Logger getInstance() {
        return instance;
    }

    // Logging method
    public void log(String message) {
        System.out.println("[LOG] " + message);
    }
}

File: Deep Skilling/Engineering concepts/Design Patterns and Principles/SingletonPatternExample/Logger.cs
using System;

namespace SingletonPatternExample
{
    public class Logger
    {
        private static readonly Logger instance = new Logger();

        private Logger()
        {
            Console.WriteLine("Logger initialized successfully.");
        }

        public static Logger GetInstance()
        {
            return instance;
        }

        public void Log(string message)
        {
            Console.WriteLine("[LOG] " + message);
        }
    }
}
